mobile/src/main/java/org/openhab/habdroid/ui/ItemPickerActivity.java

```diff
@@ -0,0 +1,277 @@
+        }
+        if (connection == null) {
+            updateViewVisibility(false, true);
+            return;
+        }
+
+        mItemPickerAdapter.clear();
+        updateViewVisibility(true, false);
+
+        final AsyncHttpClient client = connection.getAsyncHttpClient();
+        mRequestHandle = client.get("rest/items", new AsyncHttpClient.StringResponseHandler() {
+            @Override
+            public void onSuccess(String responseBody, Headers headers) {
+                try {
+                    ArrayList<Item> items = new ArrayList<>();
+                    JSONArray jsonArray = new JSONArray(responseBody);
+                    for (int i = 0; i < jsonArray.length(); i++) {
+                        JSONObject itemJson = jsonArray.getJSONObject(i);
+                        Item item = Item.fromJson(itemJson);
+                        if (!item.readOnly()) {
+                            items.add(item);
+                        }
+                    }
+                    Log.d(TAG, "Item request success, got " + items.size() + " items");
+                    mItemPickerAdapter.addLoadedItems(items);
+                    handleInitialHighlight();
+                    updateViewVisibility(false, false);
+                } catch (JSONException e) {
+                    Log.d(TAG, "Item response could not be parsed", e);
+                    updateViewVisibility(false, true);
+                }
+            }
+
+            @Override
+            public void onFailure(Request request, int statusCode, Throwable error) {
+                updateViewVisibility(false, true);
+                Log.e(TAG, "Item request failure", error);
+            }
+        });
+    }
+
+    @Override
+    public void onClick(View view) {
+        if (view == mRetryButton) {
+            loadItems();
+        }
+    }
+
+    @Override
+    public void onItemClicked(Item item) {
+        if (item == null) {
+            return;
+        }
+
+        ArrayList<String> labels = new ArrayList<>();
+        ArrayList<String> commands = new ArrayList<>();
+
+        new SuggestCommandsFactory(this, commands, labels, true).fill(item);
+
+        labels.add(getString(R.string.item_picker_custom));
+
+        final String[] labelArray = labels.toArray(new String[0]);
+        new AlertDialog.Builder(this)
+                .setTitle(R.string.item_picker_dialog_title)
+                .setItems(labelArray, (dialog, which) -> {
+                    if (which < labelArray.length - 1) {
+                        finish(true, item.name(), commands.get(which));
+                    } else {
+                        final EditText input = new EditText(this);
+                        LinearLayout.LayoutParams lp = new LinearLayout.LayoutParams(
+                                LinearLayout.LayoutParams.MATCH_PARENT,
+                                LinearLayout.LayoutParams.MATCH_PARENT);
+                        input.setLayoutParams(lp);
```

Comment:
Is this really needed? IOW: what happens when omitting this?

Reply:
Actually it does nothing, because `setLayoutParams()` must be called after `AlertDialog#show()`. And it should be `FrameLayout.LayoutParams`, otherwise the app crashes. I removed it.